File: 0007-reverse-integer/0007-reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        int revx = 0;
        while (x!= 0) {
            int lastdig = x%10;
            x = x/10;
            if (revx > INT_MAX/10 || (revx == INT_MAX / 10 && lastdig > 7)) {
                return 0;
            }
            if (revx < INT_MIN/10 || (revx == INT_MIN / 10 && lastdig < -8)) {
                return 0;
            }
            revx = revx*10 + lastdig;

        }
        return revx;
    }
};

File: 0007-reverse-integer/0007-reverse-integer.cs
public class Solution {
    public int Reverse(int x) {
        int Last_dig = 0;
        int rev = 0;

        if (x == Int32.MinValue) return 0;

        int absX = Math.Abs(x);
        
        while (absX != 0) {
            Last_dig = absX % 10;
            absX /= 10;

            if (rev > (Int32.MaxValue - Last_dig) / 10) {
                return 0;
            }

            rev = rev * 10 + Last_dig;
        }

        return x < 0 ? -rev : rev;
    }
}
